agent/engine/docker_task_engine.go

```diff
@@ -1279,12 +1280,14 @@ func getFirelensLogConfig(task *apitask.Task, container *apicontainer.Container,
 	tag := fmt.Sprintf(fluentTagDockerFormat, container.Name, taskID)
 	fluentd := socketPathPrefix + filepath.Join(cfg.DataDirOnHost, dataLogDriverPath, taskID, dataLogDriverSocketPath)
 	logConfig := hostConfig.LogConfig
+	bufferLimit := logConfig.Config[apitask.FirelensLogDriverBufferLimitOption]
 	logConfig.Type = logDriverTypeFluentd
 	logConfig.Config = make(map[string]string)
 	logConfig.Config[logDriverTag] = tag
 	logConfig.Config[logDriverFluentdAddress] = fluentd
 	logConfig.Config[logDriverAsyncConnect] = strconv.FormatBool(true)
 	logConfig.Config[logDriverSubSecondPrecision] = strconv.FormatBool(true)
+	logConfig.Config[logDriverBufferLimit] = bufferLimit
```

Comment:
this should only be added when bufferLimit is not empty

Reply:
I see here has a check for empty string so I didn't add it: https://github.com/moby/moby/blob/master/daemon/logger/fluentd/fluentd.go#L174. But yeah, we should handle this case here. Thanks for the suggestion.